yb-voyager/cmd/exportSchema.go

```diff
@@ -736,3 +757,132 @@ func clearAssessmentRecommendationsApplied() {
 		utils.ErrExit("clear assessment recommendations applied: update migration status record: %s", err)
 	}
 }
+
+func applyPerformanceOptimizationsAndGenerateReport() ([]string, error) {
+	if skipPerfOptimizations {
+		log.Infof("not applying performance optimizations due to flag --skip-performance-optimizations=true")
+		return nil, nil
+	}
+
+	if source.DBType != POSTGRESQL {
+		return nil, nil
+	}
+
+	log.Infof("applying performance optimizations to the exported schema")
+
+	//order needs to be retained atleast for a particular schema object type like INDEXES
+	//First we change the INDEXES_table.sql to indexes_before_applying_perf_optimizations.sql
+	//Then we first remove the redundant indexes from that file
+	//then will modify rest of the indexes in the file to make them range sharded indexes
+	//in case anything goes wrong, we can always refer to the indexes_before_applying_perf_optimizations.sql file
+
+	//INDEX
+	fileName := utils.GetObjectFilePath(schemaDir, INDEX)
+	if !utils.FileOrFolderExists(fileName) { // there are no indexes
+		log.Infof("INDEXES_table.sql file doesn't exists, skipping applying indexes performance optimizations")
+		return nil, nil
+	}
+
+	// copy the current INDEXES_table.sql file to indexes_before_applying_perf_optimizations.sql
+	copiedFileName := filepath.Join(filepath.Dir(fileName), "indexes_before_applying_perf_optimizations.sql")
```

Comment:
how about something to indicate that it's sort of a back up file. 
"backup.indexes_table.sql.before_applying_perf_optimizations" ?

Reply:
hmm, changed it to `backup_before_applying_perf_optimizations_INDEXES_table.sql`